src/prof.c

```diff
@@ -1709,7 +1714,70 @@ prof_dump_maps(bool propagate_err) {
 
 	cassert(config_prof);
 #ifdef __FreeBSD__
-	mfd = prof_open_maps("/proc/curproc/map");
+	int mib[4];
+	size_t sz;
+	char *buf;
+
+	mib[0] = CTL_KERN;
+	mib[1] = KERN_PROC;
+	mib[2] = KERN_PROC_VMMAP;
+	mib[3] = prof_getpid();
+	mfd = -1;
```

Comment:
Given that mfd is only used in the last `#else` case, let's refactor and only consider it there?